src/main/java/me/zeroeightsix/kami/module/modules/client/Capes.kt

```diff
@@ -22,100 +42,211 @@ import javax.net.ssl.HttpsURLConnection
+                || entity.getItemStackFromSlot(EntityEquipmentSlot.CHEST).getItem() != Items.ELYTRA) return false
+
+        val cape = capeUsers[entity.gameProfile.id]
+
+        return if (cape != null) {
+            renderElytra(renderer, model, entity, cape, limbSwing, limbSwingAmount, ageInTicks, netHeadYaw, headPitch, scale, partialTicks)
+        } else {
+            false
+        }
     }
+
+    private fun renderElytra(
+            renderer: RenderLivingBase<*>,
+            model: ModelElytra,
+            player: AbstractClientPlayer,
+            cape: Cape,
+            limbSwing: Float,
+            limbSwingAmount: Float,
+            ageInTicks: Float,
+            netHeadYaw: Float,
+            headPitch: Float,
+            scale: Float,
+            partialTicks: Float
+    ): Boolean {
+        val primaryColor = parseColor(cape.color.primary)
+        val borderColor = parseColor(cape.color.border)
+
+        if (primaryColor == null || borderColor == null) return false
+
+        renderElytraLayer(renderer, model, player, CapeTexture.PRIMARY, primaryColor, limbSwing, limbSwingAmount, ageInTicks, netHeadYaw, headPitch, scale, partialTicks)
+        renderElytraLayer(renderer, model, player, CapeTexture.BORDER, borderColor, limbSwing, limbSwingAmount, ageInTicks, netHeadYaw, headPitch, scale, partialTicks)
+
+        if (cape.type == CapeType.CONTRIBUTOR) {
+            renderElytraLayer(renderer, model, player, CapeTexture.TEXT_ICON, DyeColors.WHITE.color, limbSwing, limbSwingAmount, ageInTicks, netHeadYaw, headPitch, scale, partialTicks)
+        } else {
+            renderElytraLayer(renderer, model, player, CapeTexture.TEXT, DyeColors.WHITE.color, limbSwing, limbSwingAmount, ageInTicks, netHeadYaw, headPitch, scale, partialTicks)
+        }
+
+        return true
+    }
+
+    private fun renderElytraLayer(
+            renderer: RenderLivingBase<*>,
+            model: ModelElytra,
+            player: AbstractClientPlayer,
+            texture: CapeTexture,
+            color: ColorHolder,
+            limbSwing: Float,
+            limbSwingAmount: Float,
+            ageInTicks: Float,
+            netHeadYaw: Float,
+            headPitch: Float,
+            scale: Float,
+            partialTicks: Float
+    ) {
+        GlStateManager.color(color.r / 255.0f, color.g / 255.0f, color.b / 255.0f, 1.0f)
+        renderer.bindTexture(texture.location)
+        GlStateManager.pushMatrix()
+        GlStateManager.translate(0.0f, 0.0f, 0.125f)
+        model.setRotationAngles(limbSwing, limbSwingAmount, ageInTicks, netHeadYaw, headPitch, scale, player)
+        model.render(player, limbSwing, limbSwingAmount, ageInTicks, netHeadYaw, headPitch, scale)
+
+        if (player.getItemStackFromSlot(EntityEquipmentSlot.CHEST).isItemEnchanted) {
+            LayerArmorBase.renderEnchantedGlint(renderer, player, model, limbSwing, limbSwingAmount, partialTicks, ageInTicks, netHeadYaw, headPitch, scale)
+        }
```

Comment:
Potential enchant glint bug here. Need some test I guess. If it breaks maybe only render enchant glint for the last layer?